test/services/FinancialTransactionsServiceSpec.scala

```diff
@@ -180,4 +180,53 @@ class FinancialTransactionsServiceSpec extends SpecBase with MockFinancialDataCo
 
     }
   }
+
+  "The FinancialTransactionService.checkDirectDebitExists method" should {
+
+    val vrn = "123456"
+
+    "CheckDirectDebitExists Return true when a success response is returned from the Connector" in {
+
+      val successResponse: Either[Nothing, Boolean] = Right(true)
+      setupMockCheckDirectDebitExists(vrn)(successResponse)
+
+      setupMockAuditEventResponse(DirectDebitsCheckResponseAuditModel(vrn, hasDirectDebit = true))
+      setupMockAuditEventResponse(DirectDebitCheckRequestAuditModel(vrn))
+
+      val actual: Either[ErrorResponse, Boolean] = await(TestFinancialTransactionService.checkDirectDebitExists(vrn))
+
+      actual shouldBe successResponse
+
+      verifyAuditEvent(DirectDebitCheckRequestAuditModel(vrn))
```

Comment:
These _verifyAuditEvent_ functions don't return anything and are after the test assertion. Are they necessary?

Reply:
They verify the method was called during the test